examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -274,18 +286,18 @@ int main(int argc, char **argv)
     const std::string rngnKey = "1b GenRnGen";
     timermap.start( rngnKey );
 #ifdef MGONGPU_COMMONRAND_ONHOST
-    std::vector<double> commonRandomNumbers = commonRandomPromises[iiter].get_future().get();
-    assert( nRnarray == static_cast<int>(commonRandomNumbers.size()) );
+    std::vector<double> commonRnd = commonRandomPromises[iiter].get_future().get();
+    assert( nRnarray == static_cast<int>(commonRnd.size()) );
     // NB (PR #45): memcpy is strictly needed only in CUDA (copy to pinned memory), but keep it also in C++ for consistency
-    memcpy( hstRnarray, commonRandomNumbers.data(), nbytesRnarray );
+    std::copy( commonRnd.begin(), commonRnd.end(), hstRnarray.get() );
```

Comment:
any particular reason why std::copy is better? I think memcpy makes it more obvious that we know we are copying a memy block of adjacent stuff... but maybe I am wrong

Reply:
Of course. :-)
`nbytesRnarray` triggers a warning if specific branches of the `ifdef`s are active. It's because it was used both for cudaMemcpy and the on-host copy.
I could think of three ways to solve it:
- cast it to void in all branches where it is defined but not used. (Not nice)
- Hide behind more ifdefs. (I try to avoid as many ifdefs as I can, because you see errors only when you reconfigure. In fact, I saw the warning only when I did my last checks and cycled through the three options. I found that annoying / unsafe.)
- Use std::copy for the host copy, and cudamemcpy for a cuda memcopy. This way, `nbytesRnarray` is only used in one place.

Option 3 seemed the most obvious in C++

Reply:
ah :-) not obvious, I thought you preferred std::copy to memcpy for some other reasons

I would go for option 2, which is what was used in the previous code and was working fine without any error or warning - ie redefine rnsrray excatly in the same way behind another ifdef

i do not see any problem with having pieces of code used only when the relevant define is defined.... this is the same we discussed this morning

Reply:
> ah :-) not obvious, I thought you preferred std::copy to memcpy for some other reasons

I do that as well :-)
It doesn't ask for a size, and it checks that the types are compatible.

Reply:
Thanks for going back to memcpy and the original #ifdef :-)
I think memcpy with the explicit size makes it much clearer that these are fixed size C-like arrays, much more similar to cuda.